src/charm.py

```diff
@@ -73,6 +73,51 @@ def _ensure_service_stopped(service_name: str):
         assert not service_running(service_name)
 
 
+def _parse_x_stream_protocol_version(config_str):
+    """
+    Parse the x-stream-protocol-version configuration string.
+
+    Parameters:
+    - config_str (str): The configuration string for x-stream-protocol-version.
+
+    Returns:
+    - str: A comma-separated string of valid protocol versions.
+
+    Raises:
+    - ValueError: If any of the specified protocol versions are invalid.
+
+    Behavior:
+    - If config_str is empty, returns empty
+    - If valid values are provided, splits on space, verifies each element,
+      and applies to the header with ", ".join(valid_x_protocol_versions).
+    - If any invalid value, raises a ValueError.
+    """
+    allowed_versions = [
+        "v5.channel.k8s.io",
+        "v4.channel.k8s.io",
+        "v3.channel.k8s.io",
+        "v2.channel.k8s.io",
+        "channel.k8s.io",
+    ]
+
+    if not config_str.strip():
+        return ""
+
+    valid_versions = []
+    invalid_versions = []
+
+    for version in map(str.strip, config_str.split(",")):
```

Comment:
so, if you make the config just "space separated" this gets much simpler

```suggestion
    for version in config_str.split():
```

Reply:
hi @addyess , using comma-separated list can be consistent with kube-api and RFC9110 specification, they both use comma-separated list. Given this alignment with established standards, would it be necessary to make this change?

Reply:
Ahh, yeah i see the point.  Alright I'm good with that explanation